viewing_party/main.py

```diff
@@ -0,0 +1,237 @@
+
+def create_movie(title, genre, rating):
+    if title == None:
+        return None
+    elif genre == None:
+        return None
+    elif rating == None: 
+        return None
+    else:
+        movie_dict = {
+            'title': title, 
+            'genre': genre, 
+            'rating': rating
+        } 
+    print("movie dictionary:",movie_dict)
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+
+    '''
+    the value of `movie` will be a dictionary in this format:
+    
+      {
+        "title": "Title A",
+        "genre": "Horror",
+        "rating": 3.5
+      }
+    '''
+    #the value of `user_data` will be a dictionary \n
+    #  with a key `"watched"`, and a value `[]`
+    user_data["watched"] = []
+    user_data["watched"].append(movie)
+    return user_data
+
+# Test 3
+def add_to_watchlist(user_data,movie):
+   # movie["title"]
+    # print("movie:", movie)
+    user_data_watchlist = user_data["watchlist"]
+    # print("user")
+    user_data["watchlist"].append(movie)
+    return user_data
+#Test 4
+
+def watch_movie(user_data,title):
+    #iterate through the watchlist of the user
+    for movie in user_data["watchlist"]:
+        # print("movie:", movie)
+        if movie["title"] == title:
+            #add to watched list
+            user_data["watched"].append(movie) 
+            #remove from watch list 
+            user_data["watchlist"].remove(movie)  
+            # print(user_data)
+    return user_data
+                  
+
+
+#Wave 3 
+"""
+# - Consider the movies that the user has watched, and  consider the movies that their friends have watched. Determine which movies the user has watched, but none of their friends have watched.
+# - Return a list of dictionaries, that represents a list of movies
+"""
+def get_unique_watched(user_data):
+    if not user_data["watched"]:
+        return []
+    
+    # get movies watched by user
+    # get movies watched by friends
+    # get unique movies the user has watched but friends have not watched.  
+    # compare movies watched by friends to amanda's watched
+    user_watched_movie_titles = user_data["watched"]
+    friends_watched_movie_titles = user_data["friends"]
+    user_unique_movies_titles = []
+    # for movie in amanda_watched_movie_titles:
+       
+    #FOR name_of_individual_thing_in_array IN array_you_want_to_loop_through:
+    for title in user_watched_movie_titles: 
+        count = 0 #unique movie is Zero
+        for friend in friends_watched_movie_titles: 
+            if count > 0:
+                break 
+            if title in friend["watched"]:
+                count += 1
+        if count == 0:    
+            user_unique_movies_titles.append(title)    
```

Comment:
This totally works, but there are more succinct ways to solve this without the need for a `count` variable, since it doesn't really matter what the count is if it's above 0.
```suggestion
        for friend in friends_watched_movie_titles: 
            if title not in friend["watched"]:    
                user_unique_movies_titles.append(title)    
```